component/otelcol/processor/resourcedetection/internal/resource_attribute_config/resource_attribute_config.go

```diff
@@ -0,0 +1,28 @@
+package resource_attribute_config
+
+// Configures whether a resource attribute should be enabled or not.
```

Comment:
Note that `otelcol.processor.resourcedetection` will work differently from the Collector's original processor:
* In the Collector, many resource attributes are enabled by default. But some resource attributes are not.
* In the Agent, all resource attributes will be disabled by default. Users will have to enable them explicitly.

There are a few reasons why I decided to do it this way:
* If the Agent sets various resource attributes by default, the Collector might print warnings which are confusing to the user. This is the `enabledSetByUser` parameter mentioned in the comment below.
* So that we don't have to waste time keeping the Agent's defaults in sync with the Collector's.

For example, this is a labda resource detection with all resource attributes enabled:
```
otelcol.processor.resourcedetection "default"  {
  detectors = ["lambda"]
  lambda {
    resource_attributes {
      aws.log.group.names { enabled = true }
      aws.log.stream.names { enabled = true }
      cloud.platform { enabled = true }
      cloud.provider { enabled = true }
      cloud.region { enabled = true }
    }
  }
  output {
    logs = [otelcol.exporter.otlp.default.input]
  }
}
```

If a user wants to disable a resource attribute, all they need to do is just not list it:
```
otelcol.processor.resourcedetection "default"  {
  detectors = ["lambda"]
  lambda {
    resource_attributes {
      aws.log.group.names { enabled = true }
    }
  }
  output {
    logs = [otelcol.exporter.otlp.default.input]
  }
}
```

They could also explicitly set it to `enabled = false`:
```
otelcol.processor.resourcedetection "default"  {
  detectors = ["lambda"]
  lambda {
    resource_attributes {
      aws.log.group.names { enabled = true }
      cloud.region { enabled = false }
    }
  }
  output {
    logs = [otelcol.exporter.otlp.default.input]
  }
}
```

Reply:
I completely understand the approach you are suggesting and support it. The only concern I have is will this change prevent the `otelcol.processor.resourcedetection` component from becoming a distribution of otel collector resourcedetectionprocessor?

Reply:
@mar4uk I decided to backtrack on this, and instead I made the Agent component behave similarly to the Collector one. It will default certain resource attributes. I hope this makes things easier for users. ATM it defaults the same ones which the Collector defaults, but in the future it might diverge a bit. That should be ok.